### ISSUE TITLE
[FEATURE_REQUEST] Cascading delete

### ISSUE THREAD

**Issue by ArdentZeal** [balderdashy/waterline#251]
Hello,

just a short question regarding associations I could not find any docu to.

If I destroy an object, which has associations, does waterline update them automatically or do i have to update them manually?

Kind regards

**Comment by japel** [comment]
+1

**Comment by psychojoe** [comment]
+1

**Comment by albertkim** [comment]
+1

**Comment by devinivy** [comment]
@particlebanana @tjwebb @dmarcelino 0.11?

**Comment by dmarcelino** [comment]
I feel a bit apprehensive in adding too much stuff to https://github.com/balderdashy/waterline/milestones/0.11 as the list is getting longer and not many of those issues are getting fixed... if someone volunteers to implement this then yes, I think we can safely add it to 0.11.

**Comment by tibotiber** [comment]
+1

**Comment by jakutis** [comment]
+1

**Comment by wrightlabs** [comment]
+1

**Comment by slavafomin** [comment]
+1

**Comment by sofiaguyang** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @ArdentZeal.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Reopening until this is added to the roadmap via a PR.  Anyone, feel free!

**Comment by devinivy** [comment]
Nevermind– it's in the roadmap!

**Comment by galmet** [comment]
Hi, Sorry folks where can I see the roadmap, please? I am curious about the progress

Thanks

**Comment by slavafomin** [comment]
Software necromancers… = )

**Comment by Benew** [comment]
+1

**Comment by albertkim** [comment]
The Node ORM ecosystem has a lot of great and not great choices. What I have done for my applications is the following:

- Stopped using Waterline. There 's no need in a production application to agnostically connect to a sql and a no-sql database and treat them the same way. There are very special considerations for each.

- Go lower level. Sticking with knex will give the greatest flexibility in querying your database and organizing your business objects the way that you want. This is my current approach to my production apps.

- Or use an ORM that is actually properly supported. The only one I would consider at the moment is Objection.js. The developers actively maintain it, but for me the litmus test for a good package is when it maintain Typescript definitions internally in the package, which Objection does. As a note, I have not used Objection yet, but it looks very promising.

**Comment by Benew** [comment]
@albertkim thanks for the input it's inspiring.
Questions come up now, are you still using parts of blueprint? Do you implement some sort of global model lifecycle?

**Comment by albertkim** [comment]
I stopped using Sails in general 1.5 years ago when I started needing tighter control over my REST APIs. What has benefited me is being explicit in my code and keeping dependencies low. Because of that, I do not use the Blueprint API in Sails.

For database models, I simply use Knex queries, and wrap them in my own business objects/classes, which gives me total control over computed properties and special behaviors without being locked into anything else. The upside is that everything is hyper explicit, and as long as your code is well architected, new engineers can follow easily. The downside is that you write more code that other libraries have already tried to write for these use cases.

Everyone needs different things for their apps, and the goal of Sails is to be more configuration driven and to write less code. That might be just what you're looking for.

**Comment by galmet** [comment]
Thanks, @albertkim I  finally decided to switch from Waterline to Sequelize. So far so good .. I would have preferred to be able to use the ORM that comes together with the framework but I feel like I have no choice at this point.

**Comment by macpatel** [comment]
seriously sails looks like a bad choice to me overall. Specially the ORM, i feel paralyzed with it.

**Comment by mikermcneil** [comment]
Cascading delete has been implemented as part of Sails v1 / Waterline 0.13 (see upgrading guide for more info)
